native/core/src/execution/planner.rs

```diff
@@ -1108,6 +1108,44 @@ impl PhysicalPlanner {
                     .map(|expr| self.create_expr(expr, Arc::clone(&required_schema)))
                     .collect();
 
+                let default_values: Option<HashMap<usize, ScalarValue>> = if !scan
+                    .default_values
+                    .is_empty()
+                {
+                    // We have default values. Extract the two lists (same length) of values and
+                    // indexes in the schema, and then create a HashMap to use in the SchemaMapper.
+                    let default_values: Vec<ScalarValue> = scan
+                        .default_values
+                        .iter()
+                        .map(|expr| {
+                            let literal = self
+                                .create_expr(expr, Arc::clone(&required_schema))
+                                .unwrap();
+                            literal
+                                .as_any()
+                                .downcast_ref::<DataFusionLiteral>()
+                                .ok_or_else(|| {
+                                    GeneralError("Expected literal of default value.".to_string())
+                                })
+                                .map(|literal| literal.value().clone())
+                                .unwrap()
+                        })
+                        .collect::<Vec<ScalarValue>>();
```

Comment:
nit: we can avoid unwrap here:

```suggestion
                    let default_values: Result<Vec<ScalarValue>, DataFusionError> = scan
                        .default_values
                        .iter()
                        .map(|expr| {
                            let literal = self.create_expr(expr, Arc::clone(&required_schema))?;
                            let df_literal = literal
                                .as_any()
                                .downcast_ref::<DataFusionLiteral>()
                                .ok_or_else(|| {
                                GeneralError("Expected literal of default value.".to_string())
                            })?;
                            Ok(df_literal.value().clone())
                        })
                        .collect();
                    let default_values = default_values?;
```

Reply:
Looks good, thank you for the suggestion!